server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -261,10 +261,12 @@ private static LocalCheckpointTracker createLocalCheckpointTracker(EngineConfig
             // Thus, we need to restore the LocalCheckpointTracker bit by bit to ensure the consistency between LocalCheckpointTracker and
             // Lucene index. This is not the only solution since we can bootstrap max_seq_no_of_updates with max_seq_no of the commit to
             // disable the MSU optimization during recovery. Here we prefer to maintain the consistency of LocalCheckpointTracker.
-            if (localCheckpoint < maxSeqNo && engineConfig.getIndexSettings().isSoftDeleteEnabled()) {
+            // The max_seq_no of Lucene commit in the old indices might be smaller than seq_no of some documents in the commit.
```

Comment:
If this is the case, do we not then also risk that the commit that we use is not the safe commit? Since when we find the safe commit we use the max seq-no and take the newest one with max-seq-no <= global-checkpoint?

This change is still good, but I wonder if we should tackle that in a follow-up?

Reply:
Note that this issue only affected 6.6.0 and 6.6.1 with soft-deletes enabled (when we introduced soft-deletes, which was by default disabled in these releases, and had to be explicitly enabled for CCR indices, back when CCR was still a beta feature). I think we don't need to spend too much engineering effort on this.

Reply:
This bug also affects indices created between 7.0 and 7.2 where soft-deletes are enabled by default.

Reply:
I'm confused. I thought https://github.com/elastic/elasticsearch/pull/38879 was fixed in 7.0?

Reply:
Sorry, I was confused by the label v7.2.0 in #38879. You're right here.